### PR TITLE
Cleanup code gen for classes.

### PR DESCRIPTION
**Pull request by arv**
By first transforming a ClassDeclaration

```js
class C {}
```

to a let and ClassExpression

```js
let C = class C {};
```

We can simplify the code slightly.

Now we always end up using an IIFE which is better because using
a FunctionDeclaration has the wrong semantics.

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton PTAL

File: src/codegeneration/ClassTransformer.js [hunk 1]
```diff
@@ -193,7 +205,34 @@ export class ClassTransformer extends TempVarTransformer {
...
+   */
+  transformClassDeclaration(tree) {
+    // `class C {}` is equivalent to `let C = class C {};`
+    // Convert to class expression and transform that instead.
+    let classExpression = new ClassExpression(tree.location, tree.name,
+        tree.superClass, tree.elements, tree.annotations, tree.typeParameters);
+    let transformed = this.transformClassExpression(classExpression);
+    let useLet = !this.options_.transformOptions.blockBinding &&
+                 this.options_.parseOptions.blockBinding;
+    return createVariableStatement(useLet ? LET : VAR, tree.name, transformed);
```
Comment (johnjbarton) [inline src/codegeneration/ClassTransformer.js]: If we are not using blockBinding in transforms we useLet in the output?

File: src/codegeneration/PlaceholderParser.js [hunk 2]
```diff
@@ -260,7 +260,7 @@ export class PlaceholderTransformer extends ParseTreeTransformer {
           this.transformIdentifierExpression(tree.expression);
       if (transformedExpression === tree.expression)
         return tree;
-      if (transformedExpression.isStatement())
+      if (transformedExpression.isStatementListItem())
         return transformedExpression;
```
Comment (johnjbarton) [inline src/codegeneration/PlaceholderParser.js]: I wonder why this change impacts no tests.

**Comment by johnjbarton** [comment]
LGTM

File: src/codegeneration/ClassTransformer.js (hunk 1 above)
Comment (arv) [inline src/codegeneration/ClassTransformer.js]: If we are parsing let/const but not transforming them it is OK to output them.

File: src/codegeneration/PlaceholderParser.js (hunk 2 above)
Comment (arv) [inline src/codegeneration/PlaceholderParser.js]: FunctionDeclaration is not a Statement but a Declaration and StatementListItem is the union of those two.

Comment (arv) [inline src/codegeneration/PlaceholderParser.js]: I guess we never used FunctionDeclaration in this context before.
